src/main/scala/com/databricks/spark/csv/util/InferSchema.scala

```diff
@@ -53,14 +55,20 @@ private[csv] object InferSchema {
     StructType(structFields)
   }
 
-  private def inferRowType(nullValue: String, dateFormatter: SimpleDateFormat)
+  private def inferRowType(nullValue: String,
```

Comment:
The idea here is to know whether the user has used the DROPMALFORMED parse mode. Only in this case, the type inference should avoid taking into account the malformed lines. Whether a line is malformed or not is not the only criteria for doing the extra 'if' there. We need to check whether the user has Chosen the DROPMALFORMED Option as well. In this case, the parse mode is consistent both in the type inference and in the generation of the data Frame out of the CSV. 

This exact logic is depicted in my three unit tests as well. 

I also added a comment in the Code for avioding disambiguations.